### PR TITLE
Add Mutation Testing

### PR DESCRIPTION
**Pull request by theofidry**
**https://github.com/theofidry/mutation-testing**

**A curated list of useful resources related to mutation testing: the existing libraries across different languages, research papers, videos, conferences.**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

**Comment by sindresorhus** [comment]
Ping. Please also see #1363.

File: readme.md [hunk 1]
```diff
@@ -292,6 +292,7 @@
 - [Open Source Society University](https://github.com/ossu/computer-science#readme)
 - [Functional Programming](https://github.com/lucasviola/awesome-functional-programming#readme)
 - [Static Analysis & Code Quality](https://github.com/mre/awesome-static-analysis#readme)
+- [Mutation Testing](https://github.com/theofidry/mutation-testing)
```
Comment (janisozaur) [inline readme.md]: shouldn't in link to `readme` anchor, like all the others?

**Comment by theofidry** [comment]
It's not really needed but you're right let's be consistent

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/theofidry/awesome-mutation-testing
✖ Linting

  README.md:35:51
  ✖    1:1    Missing License section                                       remark-lint:awesome/license
  ✖    1:1    Missing or invalid Table of Contents                          remark-lint:awesome/toc
  ✖   11:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   12:1    Marker should be 2, was 1                                     remark-lint:ordered-list-marker-value
  ✖   12:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   13:1    Marker should be 3, was 1                                     remark-lint:ordered-list-marker-value
  ✖   13:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   14:1    Marker should be 4, was 1                                     remark-lint:ordered-list-marker-value
  ✖   14:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   15:1    Marker should be 5, was 1                                     remark-lint:ordered-list-marker-value
  ✖   15:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   16:1    Marker should be 6, was 1                                     remark-lint:ordered-list-marker-value
  ✖   16:4    Invalid list item link URL                                    remark-lint:awesome/list-item
  ✖   35:50   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   35:51   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   38:74   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   38:75   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   40:66   List item description must end with proper punctuation        remark-lint:awesome/list-item
  ✖   46:52   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   46:53   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   53:70   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   53:71   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   54:58   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   54:59   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   55:106  List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   55:107  Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   65:62   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   65:63   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   66:54   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   66:55   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   67:98   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   67:99   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   70:70   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   70:71   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   71:56   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   71:57   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   72:70   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   72:71   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   75:54   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   75:55   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   80:52   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   80:53   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   86:66   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   86:67   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖   87:50   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖   87:51   Found reference to undefined definition                       remark-lint:no-undefined-references
  ✖  128:1    Marker style should be *                                      remark-lint:unordered-list-marker-style
  ✖  129:3    Marker style should be *                                      remark-lint:unordered-list-marker-style
  ✖  133:1    Marker style should be *                                      remark-lint:unordered-list-marker-style

  49 errors
```

https://github.com/sindresorhus/awesome-lint

File: readme.md [hunk 2]
```diff
@@ -292,6 +292,7 @@
 - [Open Source Society University](https://github.com/ossu/computer-science#readme)
 - [Functional Programming](https://github.com/lucasviola/awesome-functional-programming#readme)
 - [Static Analysis & Code Quality](https://github.com/mre/awesome-static-analysis#readme)
+- [Mutation Testing](https://github.com/theofidry/mutation-testing#readme)
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
Include a short description about the project/theme of the list. **It should not describe the list itself.**

**Comment by sindresorhus** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)

**Comment by sindresorhus** [comment]
`Contents` should be an unordered list.

**Comment by sindresorhus** [comment]
Don't include `[abandoned]` items. This is an **Awesome** list. Abandoned stuff are not awesome.

**Comment by sindresorhus** [comment]
Ping

**Comment by theofidry** [comment]
@sindresorhus updated the requested items at the exception of:

>Don't include [abandoned] items. This is an Awesome list. Abandoned stuff are not awesome.

I have very mitigated feelings about it. Whilst I get the reasoning, in this peculiar case, mutation testing is a very green field. It's not exactly recent, but the tooling in general is very immature and it's not rare at all to still see a lot new mutation testing tools popping up in different languages. Since this list is about providing material about mutation testing, it is very interesting and important to be able to asses the existing (including abandoned tools) in order to identity what is there, missing or why it failed.

In other words: abandoned packages still can provide a lot of value for this topic.

**Comment by sindresorhus** [comment]
> it is very interesting and important to be able to asses the existing (including abandoned tools) in order to identity what is there, missing or why it failed.

None of the repos marked as `[abandoned]`, as far as I could see, includes any reasoning for why they failed, and I don't think users will or even easily can determine that. So your reasoning is not very sound unless you plan to include a note on each item about the reason they failed and what the user can learn from that. If you insist on keeping them, I would suggest moving them to a separate Markdown file with the same organization, but just for abandoned items. Right now there's just too many abandoned project in the main readme and they create too much noise.

**Comment by theofidry** [comment]
Ok, done

**Comment by sindresorhus** [comment]
> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
> - https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md

---

Header: `Awesome Mutation testing` => `Awesome Mutation Testing`

---

```
❯ awesome-lint https://github.com/theofidry/awesome-mutation-testing
✖ Linting

  README.md:26:3
  ✖    1:1   Missing License section                                                remark-lint:awesome/license
  ✖    1:1   Missing or invalid Table of Contents                                   remark-lint:awesome/toc
  ✖   11:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   12:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   13:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   14:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   15:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   16:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   17:3   Invalid list item link URL                                             remark-lint:awesome/list-item
  ✖   22:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   23:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   24:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   25:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   26:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   26:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   27:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   27:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   28:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   28:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   29:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   29:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   30:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   30:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   31:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   32:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   32:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   33:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   33:7   Link to http://sei.pku.edu.cn/%7Exiongyf04/papers/ISSTA17.pdf is dead  remark-lint:no-dead-urls
  ✖   34:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   34:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   35:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   36:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   36:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   37:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   37:3   Incorrect indentation before bullet: remove 2 spaces                   remark-lint:list-item-bullet-indent
  ✖   40:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   41:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   42:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   43:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   44:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   45:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   46:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   47:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   48:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   49:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   50:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   51:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   52:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   53:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   54:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   55:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   55:54  List item link and description must be separated with a dash           remark-lint:awesome/list-item
  ✖   57:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   58:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   59:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   60:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   61:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   62:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   63:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   64:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   65:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   66:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   67:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   68:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   69:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   70:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   71:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   72:5   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   73:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   74:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   75:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   76:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   77:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   78:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   79:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   80:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   81:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   82:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   83:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   84:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   85:3   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   89:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   94:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   95:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   96:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   97:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   98:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖   99:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  101:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  105:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  106:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  107:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  110:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  111:1   Marker style should be -                                               remark-lint:unordered-list-marker-style
  ✖  112:1   Marker style should be -                                               remark-lint:unordered-list-marker-style

  95 errors
```

https://github.com/sindresorhus/awesome-lint

File: readme.md [hunk 3]
```diff
@@ -292,6 +292,7 @@
 - [Open Source Society University](https://github.com/ossu/computer-science#readme)
 - [Functional Programming](https://github.com/lucasviola/awesome-functional-programming#readme)
 - [Static Analysis & Code Quality](https://github.com/mre/awesome-static-analysis#readme)
+- [Mutation Testing](https://github.com/theofidry/mutation-testing#readme) - Better code by making bugs.
```
Comment (sindresorhus) [inline readme.md]: The description is not really understandable without context. It should be clearer.

**Comment by AshleighAdams** [comment]
Why was this closed? I would expect the linked mutation testing doc to be linked here. I found it via Google instead, but still...

**Comment by Jogai** [comment]
Because the list is not following the awesome standards? It still could be awesome, just not up to standards to include it here.
